### PR TITLE
REF: Use existing nav ref for scanQR helper

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/czl5pboartbpgccgd7uljp43fe?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1720)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Review by limpbrains — APPROVED** [review]

File: NavigationService.ts [hunk 1]
```diff
@@ -2,9 +2,9 @@ import { createNavigationContainerRef, NavigationAction, ParamListBase } from '@
 
 export const navigationRef = createNavigationContainerRef<ParamListBase>();
 
-export function navigate(name: string, params?: ParamListBase) {
+export function navigate(name: string, params?: ParamListBase, merge?: boolean) {
   if (navigationRef.isReady()) {
-    navigationRef.current?.navigate(name, params);
+    navigationRef.current?.navigate({ key: name, params, merge });
```
Comment (Overtorment) [inline NavigationService.ts:7]: not sure if `undefined` is acceptable here, maybe make a default value for `merge` false..?

File: helpers/scan-qr.ts [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+  currentScreenName,
   showFileImportButton = true,
-  onDismiss?: () => void,
-): Promise<string | null> {
+  onDismiss,
+}: {
+  currentScreenName: string;
+  showFileImportButton?: boolean;
+  onDismiss?: () => void;
+}): Promise<string | null> {
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply helpers/scan-qr.ts:23]: Those dont have autocomplete. At least not with VS Code.

[the remark this answers is not in the document]
Comment (Overtorment) [reply helpers/scan-qr.ts:23]: smth is not right then with the IDE. webstorm autocompletes and typechecks fine

[the remark this answers is not in the document]
Comment (marcosrdz) [reply helpers/scan-qr.ts:23]: With object arg u gotta make sure u provide the values in specific order. Unlike dictionary.

[the remark this answers is not in the document]
Comment (marcosrdz) [reply helpers/scan-qr.ts:23]: You also have to know what key is issue value for from memory, unlike a dictionary where you can just see it without having to go into the function signature

File: screen/send/Broadcast.tsx [hunk 3]
```diff
@@ -1,8 +1,7 @@
-import { useNavigation, useRoute } from '@react-navigation/native';
-import * as bitcoin from 'bitcoinjs-lib';
 import React, { useState } from 'react';
+import { useRoute } from '@react-navigation/native';
+import * as bitcoin from 'bitcoinjs-lib';
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/send/Broadcast.tsx:3]: import React should be on top when its a React component

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mk37zblohc5hx5aphreemhucjq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1736)

File: helpers/scan-qr.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Overtorment) [reply helpers/scan-qr.ts:23]: image

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gc4p7aa4kvjeqctwu7zu4z3sne?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1748)

File: helpers/scan-qr.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcosrdz) [reply helpers/scan-qr.ts:23]: @Overtorment  you won. I'll close  this PR. too many changes required (including test file) and its not worth the effort.

**Pull request closed without merging**